ReadMe記述

TomoyaOkuhiro opened the pull request:
# Why 
   
 DB設計をし、円滑に実装を進めていく為。

# What

  ReadMeファイルにカラム情報を記述。

TomoyaOkuhiro committed:
ReadMe記述

kazuki04 commented on README.md:
```diff
@@ -1,24 +1,41 @@
+| Column             | Type                | Options                 |
+|--------------------|---------------------|-------------------------|
+| nickname           | string              | null: false             |
+| e-mail             | string              | null: false             |
```
```
| e-mail | string | null: false |
```
メールアドレスに関しては一意性の制約があるため、「unique: true」のオプションを追加しましょう！

TomoyaOkuhiro replied:
分かりました。修正します。

kazuki04 commented on README.md:
```diff
@@ -1,24 +1,41 @@
+|--------------------|---------------------|-------------------------|
+| nickname           | string              | null: false             |
+| e-mail             | string              | null: false             |
+| password           | string              | null: false             |
```
```
| password | string | null: false |
```
パスワードのカラムは「encrypted_password」に修正しましょう!
deviseにより、passwordのカラムに関しては、最終的にencrypted_passwordという暗号化されたものに変換されるためです。

TomoyaOkuhiro replied:
修正完了です。

kazuki04 commented on README.md:
```diff
@@ -1,24 +1,41 @@
+| first_name         | string              | null: false             |
+| last_name_kana     | string              | null: false             |
+| first_name_kana    | string              | null: false             |
+| birth_year         | string              | null: false             |
+| birth_month        | string              | null: false             |
+| birth_day          | string              | null: false             |
```
```
| birth_year         | string              | null: false             |
| birth_month        | string              | null: false             |
| birth_day          | string              | null: false             |
```
誕生日に関してですが、「年」「月」「日」と3つのカラムに分かれているのを**1つにまとめましょう。**
date型という日付に関するデータ型を使用することで実装できます。

TomoyaOkuhiro replied:
修正完了しました

kazuki04 commented on README.md:
```diff
@@ -1,24 +1,41 @@
+### Association
 
-* Configuration
+- has_many :item
```
```
has_many :item
```
アソシエーションに関しまして、has_manyのため複数形に修正しましょう！

TomoyaOkuhiro replied:
修正完了しました。

kazuki04 commented on README.md:
```diff
@@ -1,24 +1,41 @@
 
-* How to run the test suite
+| Column                              | Type       | Options           |
+|-------------------------------------|------------|-------------------|
+| name                                | string     | null: false       |
+| description                         | text       | null: false       |
+| price                               | string     | null: false       |
```
itemsテーブルに関しまして、カテゴリーなどの「プルダウンで選択をする項目」が無いかと思うため、カラムを追加しましょう！

TomoyaOkuhiro replied:
修正しましたので確認の程、よろしお願いします。

kazuki04 commented on README.md:
```diff
@@ -1,24 +1,41 @@
 
-* ...
+- belongs_to :user
+
```
テーブルに関しまして、以下の2つを用意しましょう！
```
・住所テーブル
・購入管理テーブル
```
2つとも購入時に値が保存されるテーブルとなります。
購入管理テーブルに関しましては、「誰が」「どの商品を」購入したかを管理するテーブルとなります。

kazuki04 reviewed (commented):
レビュー依頼ありがとうございます！
コメントさせていただきました。修正後、改めてレビュー依頼をお願いいたします。
また、再度レビュー依頼をしていただく際、コミットとプッシュを忘れずにお願いいたします。

TomoyaOkuhiro committed:
ReadMe 編集

Hide-Morishita commented on README.md:
```diff
@@ -1,24 +1,76 @@
+| Column             | Type                | Options                  |
+|--------------------|---------------------|--------------------------|
+| nickname           | string              | null: false              |
+| e-mail             | string              | null: false, unique: true|
```
```ruby
| e-mail | string | null: false, unique: true|
```
deviseでは、デフォルトで`email`というカラムが用意されていますので、
カラム名を修正しておきましょう。

TomoyaOkuhiro replied:
修正完了しました。

Hide-Morishita commented on README.md:
```diff
@@ -1,24 +1,76 @@
+| name                                | string     | null: false                   |
+| description                         | text       | null: false                   |
+| price                               | string     | null: false                   |
+| category                            | integer    | null: false, foreign_key: true|
+| status                              | integer    | null: false, foreign_key: true|
+| shipping_cost                       | integer    | null: false, foreign_key: true|
+| shipping_area                       | integer    | null: false, foreign_key: true|
+| shipping_day                        | integer    | null: false, foreign_key: true|
```
```ruby
| category                            | integer    | null: false, foreign_key: true|
| status                              | integer    | null: false, foreign_key: true|
| shipping_cost                       | integer    | null: false, foreign_key: true|
| shipping_area                       | integer    | null: false, foreign_key: true|
| shipping_day                        | integer    | null: false, foreign_key: true|
```
商品の状態, 配送料の負担, 発送元の地域, 発送までの日数, カテゴリーのカラムはactive_hashで実装の予定のため、**カラム名の末尾に_idを追加**しましょう。

TomoyaOkuhiro replied:
修正完了しました

Hide-Morishita commented on README.md:
```diff
@@ -1,24 +1,76 @@
 
-* Services (job queues, cache servers, search engines, etc.)
+| Column                              | Type       | Options                       |
+|-------------------------------------|------------|-------------------------------|
+| name                                | string     | null: false                   |
+| description                         | text       | null: false                   |
+| price                               | string     | null: false                   |
+| category                            | integer    | null: false, foreign_key: true|
+| status                              | integer    | null: false, foreign_key: true|
+| shipping_cost                       | integer    | null: false, foreign_key: true|
+| shipping_area                       | integer    | null: false, foreign_key: true|
+| shipping_day                        | integer    | null: false, foreign_key: true|
```
itemsテーブルはusersテーブルとのアソシエーションを結んでいますので、usersテーブルのidを参照する**外部キー**を記述しましょう。

TomoyaOkuhiro replied:
修正完了しました。

Hide-Morishita commented on README.md:
```diff
@@ -1,24 +1,76 @@
 
-* Database creation
+- has_many :items
 
```
usersテーブルは、購入管理テーブルと紐づきますので
購入管理テーブルとのアソシエーションを記述しておきましょう。

TomoyaOkuhiro replied:
修正完了しました。

Hide-Morishita commented on README.md:
```diff
@@ -1,24 +1,76 @@
+### Association
+
+- belongs_to :user
+- has_many :items
```
```ruby
has_many :items
```
こちらはitemsテーブルになりますので、itemsテーブルとのアソシエーションは不要になります。
削除しておきましょう。

itemsテーブルは、購入管理テーブルと紐づきますので、アソシエーションを記述しておきましょう。

TomoyaOkuhiro replied:
修正完了しました。

Hide-Morishita commented on README.md:
```diff
@@ -1,24 +1,76 @@
+
+## purchase_management table
+
+| Column       | Type    | Options          |
+|--------------|---------|------------------|
+| user_id      |integer  | foreign_key: true|
+
```
商品の購入管理はuserとitemのidが可能です。
「どのユーザーがどのアイテムを購入したか」を管理するテーブルになります。

TomoyaOkuhiro replied:
修正しました。

Hide-Morishita commented on README.md:
```diff
@@ -1,24 +1,76 @@
+| user_id      |integer  | foreign_key: true|
+
+has_one :address
+
```
購入管理テーブルはusersテーブル、itemsテーブルと紐づきますのでアソシエーションの記述をお願いいたします。

TomoyaOkuhiro replied:
修正しました

Hide-Morishita commented on README.md:
```diff
@@ -1,24 +1,76 @@
+| Column            | Type            | Options                       |
+|-------------------|-----------------|-------------------------------|
+| user_id           | integer         | null: false, foreign_key: true|
+| post_code         | integer         | null: false, foreign_key: true|
```
```ruby
| post_code | integer | null: false, foreign_key: true|
```
郵便番号はハイフンを必要とすることも考慮してstring型で実装しましょう！
また、`foreign_key: true`は不要ですので、削除しておきましょう。
郵便番号は、integer型にしてしまうとハイフンより前の3桁までしかDBに登録されないようになってしまいます。

TomoyaOkuhiro replied:
修正完了しました。

Hide-Morishita commented on README.md:
```diff
@@ -1,24 +1,76 @@
+| city              | string          | null: false                   |
+| address           | string          | null: false                   |
+| building_name     | string          |                               |
+| phone_number      | integer         |                               |
```
```ruby
| phone_number | integer | |
```
電話番号ですが、integer型に指定すると仮に「09012345678」を保存しようとした場合、先頭の0が保存されないので、string型に修正しましょう。
また、Optionsに`null: false`を記述しておきましょう。

TomoyaOkuhiro replied:
修正完了しました。

Hide-Morishita commented on README.md:
```diff
@@ -1,24 +1,76 @@
+
+| Column            | Type            | Options                       |
+|-------------------|-----------------|-------------------------------|
+| user_id           | integer         | null: false, foreign_key: true|
```
```ruby
| user_id | integer | null: false, foreign_key: true|
```
今回furimaでは住所情報は購入の際に記述されます。
そのため、1つの購入情報につき、1つの住所情報を持つという関係が成り立つため、usersテーブルではなくpurchase_managementテーブルとアソシエーションの関係で結びつきます。
そのため、usersテーブルのidを参照する外部キーカラムではなく、購入管理テーブル(purchase_managementテーブル)のidを参照する外部キーカラムを記述しましょう。

TomoyaOkuhiro replied:
修正完了しました

masayaomoto replied:
https://github.com/TomoyaOkuhiro/furima-33856/pull/1/files#r624424817

上記コメントの修正が反映されておりませんのでご確認お願いいたします。

Hide-Morishita commented on README.md:
```diff
@@ -1,24 +1,76 @@
+
+### Association
+
+- has_many :items
+- has_many :users
```
```ruby
has_many :items
has_many :users
```
今回furimaでは、1つの購入情報につき、1つの住所情報を持つという関係が成り立つため、usersテーブルやitemsテーブルとは、直接結びつきません。
userやitemではなく、`purchase_management`とのアソシエーションを記述しましょう。

TomoyaOkuhiro replied:
修正完了しました。

Hide-Morishita reviewed (commented):
レビュー依頼ありがとうございます。
修正点確認させていただきました。
追加でコメントいたしましたので、ご確認お願いいたします。
またトレロの要件にございます通り、「ER図」のご提出をお願いできればと存じます。

TomoyaOkuhiro committed:
ReadMe記述、編集

masayaomoto commented on README.md:
```diff
@@ -1,24 +1,83 @@
+| first_name         | string              | null: false              |
+| last_name_kana     | string              | null: false              |
+| first_name_kana    | string              | null: false              |
+| birth_day          | date                | null: false              |
```
emailを管理するカラムを追加しましょう。

masayaomoto commented on README.md:
```diff
@@ -1,24 +1,83 @@
 
-* Database creation
+- has_many :items
+- belongs_to :purchase_management
```
```
- belongs_to :purchase_management
```
ユーザーは何度でも購入をすることが可能ですので、
ユーザーテーブルと購入管理テーブルとの関係性は1対多となります。
has_manyでアソシエーションを組みましょう。

masayaomoto commented on README.md:
```diff
@@ -1,24 +1,83 @@
-* ...
+| Column                              | Type       | Options                       |
+|-------------------------------------|------------|-------------------------------|
+| name_id                             | string     | null: false, foreign_key: true|
```
```
| name_id | string | null: false, foreign_key: true|
```
こちらは商品名を管理するカラムでしょうか？
商品名に外部キー制約は不要ですので削除しましょう。

masayaomoto commented on README.md:
```diff
@@ -1,24 +1,83 @@
+|-------------------------------------|------------|-------------------------------|
+| name_id                             | string     | null: false, foreign_key: true|
+| description                         | text       | null: false                   |
+| price                               | string     | null: false                   |
```
```
| price | string | null: false |
```
こちらのカラムですが、
数値で保存するためinteger型にしましょう。

masayaomoto commented on README.md:
```diff
@@ -1,24 +1,83 @@
+| name_id                             | string     | null: false, foreign_key: true|
+| description                         | text       | null: false                   |
+| price                               | string     | null: false                   |
+| category                            | integer    | null: false, foreign_key: true|
+| status_id                           | integer    | null: false, foreign_key: true|
+| shipping_cost_id                    | integer    | null: false, foreign_key: true|
+| shipping_area_id                    | integer    | null: false, foreign_key: true|
+| shipping_day_id                     | integer    | null: false, foreign_key: true|
```
```
| category                            | integer    | null: false, foreign_key: true|
| status_id                           | integer    | null: false, foreign_key: true|
| shipping_cost_id                    | integer    | null: false, foreign_key: true|
| shipping_area_id                    | integer    | null: false, foreign_key: true|
| shipping_day_id                     | integer    | null: false, foreign_key: true|
```
商品の状態・配送料の負担・発送元の地域・発送までの日数・カテゴリー
に関してのカラムはactive_hashを用いての実装となります。
active_hashを用いたカラムのカラム名は```単数形_id```としましょう。
また、外部キー制約は不要ですので削除しましょう。

masayaomoto commented on README.md:
```diff
@@ -1,24 +1,83 @@
+| shipping_cost_id                    | integer    | null: false, foreign_key: true|
+| shipping_area_id                    | integer    | null: false, foreign_key: true|
+| shipping_day_id                     | integer    | null: false, foreign_key: true|
+
```
商品テーブルにユーザーテーブルの外部キーを追加しましょう。

masayaomoto commented on README.md:
```diff
@@ -1,24 +1,83 @@
+### Association
+
+- belongs_to :user
+- belongs_to :purchase_management
```
```
- belongs_to :purchase_management
```
「1つの商品は1つの購入情報を持っている」という設計にしたいため、
has_oneを使用して購入管理テーブルとのアソシエーションを組みましょう。

masayaomoto commented on README.md:
```diff
@@ -1,24 +1,83 @@
+|-------------------|-----------------|-------------------------------|
+| user_id           | integer         | null: false, foreign_key: true|
+| post_code         | string          | null: false                   |
+| prefecture        | string          | null: false                   |
```
```
| prefecture | string | null: false |
```
都道府県カラムはactive_hashを用いての実装となります。
active_hashを用いたカラムはinteger型にしましょう。

masayaomoto commented on README.md:
```diff
@@ -1,24 +1,83 @@
+
+### Association
+
+- has_one :purchase_management
```
```
- has_one :purchase_management
```
今回の設計では「配送先住所は1つの購入情報に属している」という設計にしたいため、
belongs_toを使用して購入管理テーブルとのアソシエーションを組みましょう。

masayaomoto commented on README.md:
```diff
@@ -1,24 +1,83 @@
+
+| Column                 | Type    | Options          |
+|------------------------|---------|------------------|
+| purchase_management_id | integer | foreign_key: true|
```
```
| purchase_management_id | integer | foreign_key: true|
```
ユーザーテーブルの外部キーに修正しましょう。

masayaomoto commented on README.md:
```diff
@@ -1,24 +1,83 @@
+
+has_one :address
+belongs_to :user
+has_many :items
```
```
has_many :items
```
「1つの購入情報は1つの商品に所属している」という設計にしたいため、
belongs_toを使用して商品テーブルとのアソシエーションを組みましょう。

masayaomoto reviewed (commented):
レビュー依頼ありがとうございます！
追加でコメントいたしましたので、ご確認お願いします！

また、前回のコメントでもございますがER図も作成しましょう。
ER図が書けましたら、次回レビュー依頼していただく際にキャプチャ画像をご提出ください。
よろしくお願いします！

TomoyaOkuhiro committed:
ReadMeの修正、ER図の作成

TomoyaOkuhiro committed:
ER図作成

https://i.gyazo.com/5d2c3f24f6fb6cb669cc05090cfb425f.png

画像ファイルになりますので、確認お願いします

Saka-toku commented on README.md:
```diff
@@ -1,24 +1,85 @@
+| Column             | Type                | Options                  |
+|--------------------|---------------------|--------------------------|
+| nickname           | string              | null: false              |
+| email              | string              | null: false              |
```
同じメールアドレスは保存できないように、一意性制約のオプションを追加しましょう！

Saka-toku commented on README.md:
```diff
@@ -1,24 +1,85 @@
 
-* Database creation
+- has_many :items
+- has_many :purchase_management
```
複数形のpurchase_managementsに修正をお願いいたします。

TomoyaOkuhiro replied:
完了しました。

Saka-toku commented on README.md:
```diff
@@ -1,24 +1,85 @@
+| name_id                             | string     | null: false                   |
+| description                         | text       | null: false                   |
+| price                               | integer    | null: false                   |
+| category                            | integer    | null: false                   |
```
categoryもアクティブハッシュ のため、適切なカラム名に修正しましょう！

TomoyaOkuhiro replied:
OKです

Saka-toku commented on README.md:
```diff
@@ -1,24 +1,85 @@
-* ...
+| Column                              | Type       | Options                       |
+|-------------------------------------|------------|-------------------------------|
+| name_id                             | string     | null: false                   |
```
商品名は文字のため「_id」の削除をお願いいたします。

TomoyaOkuhiro replied:
完了しました。！

Saka-toku commented on README.md:
```diff
@@ -1,24 +1,85 @@
+
+
+
+## address table
```
テーブル名は複数形のaddressesにしておきましょう！

TomoyaOkuhiro replied:
OKです

Saka-toku commented on README.md:
```diff
@@ -1,24 +1,85 @@
+
+| Column                           | Type            | Options                       |
+|----------------------------------|-----------------|-------------------------------|
+| purchase_management_id           | integer         | null: false, foreign_key: true|
```
他テーブルの外部キーの記述に合わせ`| purchase_management | references |`に修正をお願いいたします。

TomoyaOkuhiro replied:
OKです

Saka-toku commented on README.md:
```diff
@@ -1,24 +1,85 @@
+|----------------------------------|-----------------|-------------------------------|
+| purchase_management_id           | integer         | null: false, foreign_key: true|
+| post_code                        | string          | null: false                   |
+| prefecture                       | integer         | null: false                   |
```
`| prefecture | integer | null: false |`
配送先のカラムは、itemsテーブルの発送元の地域のカラムと同じものを使用しましょう！
今回はアクティブハッシュを使用し都道府県情報をmodelファイルに用意します。
そしてmodelファイルを呼び出せば、複数の箇所で都道府県情報が使用できるようになるためです。

TomoyaOkuhiro replied:
OKです

Saka-toku commented on README.md:
```diff
@@ -1,24 +1,85 @@
+
+
+
+## purchase_management table
```
テーブル名は複数形のpurchase_managementsに修正をお願いいたします。

TomoyaOkuhiro replied:
完了しました。

Saka-toku commented on README.md:
```diff
@@ -1,24 +1,85 @@
+
+| Column                 | Type    | Options          |
+|------------------------|---------|------------------|
+| user_id                | integer | foreign_key: true|
+| item_id                | integer | foreign_key: true|
```
itemsテーブルに合わせ以下のように修正しましょう！
```
| user                | references | foreign_key: true|
| item                | references | foreign_key: true|
```

TomoyaOkuhiro replied:
完了しました。

Saka-toku commented on README.md:
```diff
@@ -1,24 +1,85 @@
+
+has_one :address
+belongs_to :user
+belongs_to :items
```
アソシエーションの記述を、単数形のitemに修正をお願いいたします。

TomoyaOkuhiro replied:
完了しました。

Saka-toku reviewed (commented):
レビュー依頼ありがとうございます！
修正点確認させていただきました。
コメントいたしましたので、ご確認お願いいたします！

TomoyaOkuhiro committed:
ReadMe編集

yuri240 commented on README.md:
```diff
@@ -1,24 +1,85 @@
+|----------------------------------|-----------------|-------------------------------|
+| purchase_management              | references      | null: false, foreign_key: true|
+| post_code                        | string          | null: false                   |
+| items_id                         | references      | null: false, foreign_key: true|
```
```
| items_id                         | references      | null: false, foreign_key: true|
```
Addressesテーブルに商品情報を紐づける必要はないため
56行目のカラムは削除しておきましょう！

yuri240 commented on README.md:
```diff
@@ -1,24 +1,85 @@
+| purchase_management              | references      | null: false, foreign_key: true|
+| post_code                        | string          | null: false                   |
+| items_id                         | references      | null: false, foreign_key: true|
+| city                             | string          | null: false                   |
```
Addressesテーブルに**都道府県のカラム**を設けましょう！

また、こちらはアクティブハッシュのデータとして
使い回すことが可能ですので
itemsテーブルの都道府県カラムと同じものを使用しましょう！

yuri240 reviewed (commented):
レビュー依頼ありがとうございます！
修正点確認させていただきました。

追加で２点コメントさせていただきましたので、ご確認お願いします。

また、こちらのデータベース設計は
完成間近と思われますので
次回ご提出の際には**最新状態のER図のキャプチャ画像**
を添付していただけますでしょうか？

お手数ではありますが、何卒よろしくお願いいたします。

TomoyaOkuhiro committed:
ReadMe 編集

https://i.gyazo.com/caa967af29d79a60ba6c48c3511bcd77.png

ER図になります。
ご確認お願いします

Saka-toku commented on README.md:
```diff
@@ -1,24 +1,83 @@
+| Column             | Type                | Options                  |
+|--------------------|---------------------|--------------------------|
+| nickname           | string              | null: false              |
+| email              | string              | null: false,              |
```
同じメールアドレスは保存できないように、一意性制約のオプションを追加しましょう！

TomoyaOkuhiro replied:
OKです

TomoyaOkuhiro replied:
unique: true をオプションで追加しました。

Saka-toku commented on README.md:
```diff
@@ -1,24 +1,83 @@
+| post_code                        | string          | null: false                   |
+| phone_number                     | string          | null: false                   |
+| shipping_area_id                 | integer         | null: false                   |
+
```
addressesテーブルにはあと3つカラムが必要ですので、見本アプリを参考に記述をお願いいたします！

TomoyaOkuhiro replied:
OKです

Saka-toku reviewed (commented):
レビュー依頼ありがとうございます！
修正点確認させていただきました。
追加でコメントいたしましたので、ご確認お願いいたします！
またER図の確認を行いますのでER図のご提出をお願いいたします。

TomoyaOkuhiro committed:
DB設計、編集

TomoyaOkuhiro commented:
ER図データ

https://i.gyazo.com/caa967af29d79a60ba6c48c3511bcd77.png

CHURO-MAU commented on README.md:
```diff
@@ -1,24 +1,86 @@
+| shipping_area_id                 | integer         | null: false                   |
+| city                             | string          | null: false                   |
+| address                          | string          | null: false                   |
+| building_name                    | string          | null: false                   |
```
６０行目
今回のアプリでは「建物名」は入力任意項目のため
null: falseは削除しておきましょう。

TomoyaOkuhiro replied:
null: falseオプション　消去しました。

CHURO-MAU reviewed (commented):
レビュー依頼ありがとうございます！

修正点確認させていただきました。

１点追加でコメントいたしましたので、ご確認お願いいたします。

TomoyaOkuhiro committed:
ReadMe 調整

CHURO-MAU commented:
レビュー依頼ありがとうございます！
LGTMとさせていただきます。
DB設計たいへんお疲れ様でした。

マージ後、ローカルのmasterブランチに忘れずにプルしましょう。
その後、新しいブランチを作成し次の実装に取り組んでいただけますと幸いです。
これから一緒に実装頑張っていきましょう。
応援しております！
